src/Registry.php

```diff
@@ -508,11 +508,13 @@ public function set($path, $value, $separator = null)
 		switch (true)
 		{
 			case (is_object($node)):
-				$result = $node->{$nodes[$i]} = $value;
+				$result = $node->{$nodes[$i]} ?? $value;
```

Comment:
If the intent is to return the previously set value, I'd suggest returning null would be the correct behavior.  Otherwise now you're in a mixed case where if it wasn't previously set you're getting the newly set value and if it was previously set you're getting the old value and that API inconsistency isn't necessarily a good thing.

Reply:
You are right. But seems that $result = $node->{$nodes[$i]} ?? null cannot pass unit test...strange

Reply:
Which is why I said this had to be fixed on the 2.0 branch only.  Because the current code behavior, and inherently the tests, all return the newly set value and if the intent is to change it to follow the docs and return the old value, the tests need updating to account for that change in behavior.

Reply:
I see. Thank you for your patience.